File: backend/src/main/java/es/dawgrupo2/zendashop/extendedDTO/OpinionExtendedDTO.java
package es.dawgrupo2.zendashop.extendedDTO;

import es.dawgrupo2.zendashop.basicDTO.GarmentBasicDTO;
import es.dawgrupo2.zendashop.basicDTO.UserBasicDTO;

public record OpinionExtendedDTO(
    Long id,
    String comment,
    Integer rating,
    UserBasicDTO user,
    GarmentBasicDTO garment
) {
}

File: frontend/app/dtos/OpinionExtendedDTO.ts
import type GarmentBasicDTO from "./GarmentBasicDTO";
import type UserBasicDTO from "./UserBasicDTO";

export default interface OpinionExtendedDTO {
  id: number;
  comment: string;
  rating: number;
  user: UserBasicDTO;
  garment: GarmentBasicDTO;
}
